aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/CreateHandler.java

```diff
@@ -40,13 +40,18 @@
         final String awsAccountId = request.getAwsAccountId();
         final ResourceModel model = request.getDesiredResourceState();
         final String pgName = model.getProfilingGroupName();
+        final String computePlatform = model.getComputePlatform();
 
-        CreateProfilingGroupRequest createProfilingGroupRequest = CreateProfilingGroupRequest.builder()
+        CreateProfilingGroupRequest.Builder createProfilingGroupRequestBuilder = CreateProfilingGroupRequest.builder()
             .profilingGroupName(pgName)
-            .clientToken(request.getClientRequestToken())
-            .build();
+            .clientToken(request.getClientRequestToken());
+
+        if (computePlatform != null) {
+            createProfilingGroupRequestBuilder.computePlatform(computePlatform);
+        }
```

Comment:
What happens if you pass *null* to the request? If nothing changes you could pass directly the value.

Or consider initializing with the default first and then use directly the *build()* from the *Builder*.

```
String computePlatform = (model.getComputePlatform() == null) ? "Default" : model.getComputePlatform();

CreateProfilingGroupRequest createProfilingGroupRequest = ... 
    .computePlatform(computePlatform)
    .build();
```

Reply:
SDK handles `null` case for non-required fields, so we don't need to have a null check here. Removed.

> 1. I've left some minor comments, change looks good.
> 2. Please take a look at the failed build from CodeBuild, it looks like _aws-cloudformation-rpdk-java-plugin_.

For point 2: I have fixed it. The issue was because of older _aws-cloudformation-rpdk-java-plugin_ version. Updated to `2.0.0` as suggested.

Reply:
> What happens if you pass _null_ to the request? If nothing changes you could pass directly the value.
> 
> Or consider initializing with the default first and then use directly the _build()_ from the _Builder_.
> 
> ```
> String computePlatform = (model.getComputePlatform() == null) ? "Default" : model.getComputePlatform();
> 
> CreateProfilingGroupRequest createProfilingGroupRequest = ... 
>     .computePlatform(computePlatform)
>     .build();
> ```

SDK handles null case for non-required fields, so we don't need to have a null check here. Removed.